src/test/java/test/retryAnalyzer/github1519/TestClassSample.java

```diff
@@ -0,0 +1,17 @@
+package test.retryAnalyzer.github1519;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.collections.Lists;
+
+import java.util.List;
+
+public class TestClassSample {
+    static boolean retry = false;

```

Comment:
You can manage the retry logic here because the test assertion is done on the messages

Reply:
This boolean is flipped by the `RetryAnalyzer`. So the first time, the test would fail, and the control would go to `RetryAnalyzer` after which the boolean gets set to `true` by it. So the next run of the `@Test` would pass. I did that to avoid having to setup an exit criteria in the `RetryAnalyzer` implementation.

Reply:
Ok, it works too :)